### PR TITLE
chore: es6 unmarshaller

### PR DESCRIPTION
**Pull request by grant**

### CODE REVIEW

File: lib/bindings/http/unmarshaller.js [hunk 1]
```diff
@@ -47,32 +39,39 @@ function resolveBindingName(payload, headers) {
...
-      validateArgs(payload, headers);
+  unmarshall(payload, headers) {
+    return new Promise((resolve, reject) => {
+      try {
+        if (!payload) {
+          throw new TypeError("payload is null or undefined");
+        }
+        if (!headers) {
+          throw new TypeError("headers is null or undefined");
+        }
```
Comment (lance) [inline lib/bindings/http/unmarshaller.js:55]: I like that you have removed the `validateArgs()` function, but I don't think there's actually a reason to throw here. A simpler version of what you have might look like this. 

```suggestion
  unmarshall(payload, headers) {
    return new Promise((resolve, reject) => {
      if (!payload) {
        return reject(new TypeError("payload is null or undefined"));
      }
      if (!headers) {
        return reject(new TypeError("payload is null or undefined"));
      }
```
I don't think the try/catch should be there at all, actually. Its only purpose is to catch this exception and then reject the promise. If something else, e.g. `resolveBindingName()` were to throw, the caller's catch block on the thennable will handle it.

Comment (grant) [reply lib/bindings/http/unmarshaller.js:55]: Thanks. I agree, but I'm just trying to refactor without modifying the functionality.
I propose we just approve this PR and can see about removing the existing try/catches here and elsewhere.

Comment (lance) [reply lib/bindings/http/unmarshaller.js:55]: OK - just to be clear though, this suggested modification isn't changing functionality. It's just a minor optimization similar to your removal of the `validateArgs()` function.

Comment (grant) [reply lib/bindings/http/unmarshaller.js:55]: OK, I can commit your suggested change but I think it's going to create a bad PR, where it was an OK PR before.

**Comment by grant** [comment]
I added the suggested changes after you insisted, but they break the build.

**Comment by lance** [comment]
You could have made the change locally and easily seen the syntax error and then pushed them. Were you trying to prove a point? I didn't insist. I just suggested.

**Comment by grant** [comment]
> You could have made the change locally and easily seen the syntax error and then pushed them. Were you trying to prove a point? I didn't insist. I just suggested.

The suggested change doesn't just introduce format errors which are easy to fix. It introduces a rogue "catch" statement. I don't understand what we're trying to do. I'm just trying to change the unmarshaller to es6, not change the file in this small PR which might require the tests to change.

Can we please approve the PR assuming we will revert the last commit?

**Comment by grant** [comment]
There's also an error with a repeated "payload is null or undefined" error.

**Comment by grant** [comment]
OK, I fixed the error. The tests now pass locally.

**Comment by lance** [comment]
Grant, in a spirit of collaboration, I merely made a suggestion. Your veiled hostility is unwelcome.

**Comment by grant** [comment]
OK, I just don't want suggestions of breaking the PR as it was perfectly fine before. There were 2 issues with the suggestion. It's hard enough to make PRs to this repo as is.

**Comment by lance** [comment]
And yet you continue with the finger pointing.

**Comment by grant** [comment]
I'm trying to fix the DCO error, I think the commit from GitHub with the suggested change doesn't isn't signed off and is breaking DCO.

**Pull request merged**